docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+
+### About S3 Output Module
+--------------------------
+
+Purpose of S3Output module is to upload files/directories into an Amazon S3 bucket using the multipart upload feature.
+
+S3Output module is **fault-tolerant**, **statically/dynamically partitionable** and has **exactly once** semantics.
+
+Module class is **S3OutputModule** located in the package **org.apache.apex.malhar.lib.fs.s3**; please refer to [github URL](https://github.com/apache/apex-malhar/blob/master/library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3OutputModule.java).
+
+### About the operators in S3OutputModule
+-----------------------------------------
+
+Following diagram illustrates the DAG in this module:
+
+![](images/s3output/s3outputmodule.png)
+
+-   ***S3InitateFileUploadOperator***
+    -   Initiate the upload for the file using **AmazonS3Client.initiateMultipartUpload(...)** method only if the number of blocks for a file is greater than 1. By Successfully initiating the file, S3 returns a response of type **InitiateMultipartUploadResult**, which includes the **upload ID**, which is the unique identifier for the multipart upload. This **upload ID** must include whenever upload the blocks, complete an upload of a file.
+If the file has single block then the operator emits an empty **upload ID**, this is an indication of not to use the multi-part feature for the downstream operators.
+    -   This operator emits the **(filemetadata, uploadId)** to **S3FileMerger** and **(filePath, metadata, uploadId)** to **S3BlockUpload**.
+
+-   ***S3BlockUploadOperator***
+    -   **S3BlockUpload** operator upload the blocks into S3 using different API's which depends on number of blocks of a file.
+If the file has single block then upload the block into S3 using **AmazonS3Client.putObject(...)** API. S3 returns a response of type **PutObjectResult** includes the **ETag**.
+If the file has more blocks then upload the block using **AmazonS3Client.uploadPart(...)** method. S3 returns a response of type UploadPartResult includes the **ETag**. This **ETag** value must be include in request to complete multipart upload.
+    -   **S3BlockUpload** emits **(path, ETag)** to **s3FileMerger**.
+
+-   ***S3FileMergerOperator***
+    -   Complete multipart upload request using **AmazonS3Client.completeMultipartUpload(...)** and must include the **upload ID**, a list of both part numbers and corresponding **ETag** values. **S3FileMergerOperator** send the complete multi-part upload request to S3 once it have all the **part ETag's** of a file. 
+**Amazon S3** creates an object by concatenating the parts in ascending order based on part number. After a successful upload request, the parts no longer exist and S3 response includes an **ETag** which uniquely identifies the combined object data. 
+
+### Configuration Parameters
+-----------------------------
+
```

Comment:
It would be helpful to include examble values for each of these parameters (use XXX or something similar for the secret keys).